packages/compass-aggregations/src/modules/pipeline-builder/pipeline-preview-manager.ts

```diff
@@ -0,0 +1,108 @@
+import type { DataService } from 'mongodb-data-service';
+import type { AggregateOptions, Document } from 'mongodb';
+import { aggregatePipeline } from '../../utils/cancellable-aggregation';
+import { cancellableWait } from '../../utils/cancellable-promise';
+
+/**
+ * Ops that must scan the entire results before moving to the
+ * next stage.
+ */
+const FULL_SCAN_OPS = ['$group', '$bucket', '$bucketAuto'];
+
+/**
+ * Stage operators that are required to be the first stage.
+ */
+const REQUIRED_AS_FIRST_STAGE = [
```

Comment:
Same here (but 5 years), do we want to change the list?

Reply:
Yeah, moving both of these there makes sense to me, great suggestion, I'll do that!

Reply:
Okay, there are also these merge / out stages that we probably also want to derive from these constants, I'll open a separate ticket to clean this up as we are planning to do some work around `ace-autocompleter` package anyway in this project and do all of these in one go

Reply:
yeah, that would be awesome! there is already something for `$out` and `$merge` there https://github.com/mongodb-js/ace-autocompleter/blob/main/lib/constants/stage-operators.js#L487

Reply:
Oh perfect! Opened COMPASS-6181 and this is basically what I'm planning on working on next, so will refactor this pretty soon